chromium_src/tools/grit/preprocess_if_expr.py

```diff
@@ -0,0 +1,126 @@
+                   `bar.css` ==> `bar-chromium.css`
+    """
+    name_bits = os.path.splitext(file_name)
+    return "".join([name_bits[0], "-chromium", name_bits[1]])
+
+
+def purge_overrides(out_folder, in_files):
+    """Deletes all overridden upstream files from the preprocess directory - we
+       need this so removing an override doesn't break the build."""
+    for file in in_files:
+        overridden_name = get_overridden_file_name(file)
+        overridden_path = os.path.join(out_folder, overridden_name)
+        if os.path.exists(overridden_path):
+            os.remove(overridden_path)
+
+
+def maybe_keep_upstream_version(override_in_folder, out_folder, override_file):
+    """Decides whether we should keep the upstream version of a file by looking
+       at the override and see if it references the upstream version"""
+    overridden_name = get_overridden_file_name(override_file)
+
+    # The path to the brave-core override file
+    override_in_path = os.path.join(override_in_folder, override_file)
+
+    # The path to the upstream file
+    upstream_file_path = os.path.join(out_folder, override_file)
+
+    with open(override_in_path) as f:
+        text = f.read()
+
+        # If we reference the upstream file in our overridden file, make sure
+        # we keep it around.
+        if os.path.basename(overridden_name).replace('.ts', '.js') in text:
+            shutil.copy(upstream_file_path,
+                        os.path.join(out_folder, overridden_name))
+
+
+def get_brave_overrides(in_folder, in_files):
+    """Gets all the overrides we have in brave-core for this target"""
+    override_files = []
+    for file in in_files:
+        in_file = get_chromium_src_override(os.path.join(in_folder, file))
+        if os.path.exists(in_file):
+            override_files.append(file)
+
+    return override_files
+
+
+# Used to indicate that there are no overrides, so we don't need to reprocess
+class NoOverridesException(Exception):
+    pass
+
+
+@override_utils.override_function(globals())
+def main(original_function, argv):
+    # Run the preprocessor as normal, to preprocess the upstream files.
+    original_function(argv)
+
+    manifest_data = {}
+    manifest_path = None
+
+    # When we rerun the processor, we detect files which should be overridden.
+    @override_utils.override_method(argparse.ArgumentParser)
+    # pylint: disable=unused-variable
+    def parse_args(self, original_method, argv):
+        nonlocal manifest_path, manifest_data
+        args = original_method(self, argv)
+
+        cwd = os.getcwd()
+        out_folder = os.path.normpath(os.path.join(cwd, args.out_folder))
+        in_folder = os.path.normpath(os.path.join(cwd, args.in_folder))
+        override_root_folder = get_chromium_src_override(in_folder)
+
+        purge_overrides(out_folder, args.in_files)
+        overrides = get_brave_overrides(in_folder, args.in_files)
+
+        if len(overrides) == 0:
+            raise NoOverridesException
```

Comment:
let's add a comment here.
also `raise should receive an instance, not a class type.
```
if not overrides:
    # Throw an exception to abort the second call to `main()` early if no overrides were found.
    raise NoOverridesException()
```